compiler/ast/src/program/mod.rs

```diff
@@ -24,14 +24,16 @@ pub use program_scope::*;
 
 use leo_span::{Span, Symbol};
 
-use crate::Stub;
+use crate::{Module, Stub};
 use indexmap::IndexMap;
 use serde::{Deserialize, Serialize};
 use std::fmt;
 
 /// Stores the Leo program abstract syntax tree.
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct Program {
+    /// A map from module paths to module definitions.
+    pub modules: IndexMap<Vec<Symbol>, Module>,
```

Comment:
An interesting thing about this AST definition is that you could define multiple program scopes in a single project and have them all share the same module code. 
I like that possibility :)